proto/gov/v1beta/query.proto

```diff
@@ -1,14 +1,181 @@
 
 // Query defines the gRPC querier service.
 service Query {
-    // this line is used by starport scaffolding # 2
+    rpc Proposal(QueryProposalRequest) returns (QueryProposalResponse) {
+        option (google.api.http).get =
+            "/heimdall/gov/v1beta/proposals/{proposal_id}";
+    }
+
+    // Proposals queries all proposals based on given status.
+    rpc Proposals(QueryProposalsRequest) returns (QueryProposalsResponse) {
+        option (google.api.http).get = "/heimdall/gov/v1beta/proposals";
+    }
+
+    // Vote queries voted information based on proposalID, voterAddr.
+    rpc Vote(QueryVoteRequest) returns (QueryVoteResponse) {
+        option (google.api.http).get =
+            "/heimdall/gov/v1beta/proposals/{proposal_id}/votes/{voter}";
+    }
+
+    // Votes queries votes of a given proposal.
+    rpc Votes(QueryVotesRequest) returns (QueryVotesResponse) {
+        option (google.api.http).get =
+            "/heimdall/gov/v1beta/proposals/{proposal_id}/votes";
+    }
+
+    // Params queries all parameters of the gov module.
+    rpc Params(QueryParamsRequest) returns (QueryParamsResponse) {
+        option (google.api.http).get =
+            "/heimdall/gov/v1beta/params/{params_type}";
+    }
+
+    // Deposit queries single deposit information based proposalID, depositAddr.
+    rpc Deposit(QueryDepositRequest) returns (QueryDepositResponse) {
+        option (google.api.http).get =
+            "/heimdall/gov/v1beta/proposals/{proposal_id}/deposits/{depositor}";
+    }
+
+    // Deposits queries all deposits of a single proposal.
+    rpc Deposits(QueryDepositsRequest) returns (QueryDepositsResponse) {
+        option (google.api.http).get =
+            "/heimdall/gov/v1beta/proposals/{proposal_id}/deposits";
+    }
+
+    // TallyResult queries the tally of a proposal vote.
+    rpc TallyResult(QueryTallyResultRequest)
+        returns (QueryTallyResultResponse) {
+        option (google.api.http).get =
+            "/heimdall/gov/v1beta/proposals/{proposal_id}/tally";
+    }
+}
+
+// QueryProposalRequest is the request type for the Query/Proposal RPC method.
+message QueryProposalRequest {
+    // proposal_id defines the unique id of the proposal.
+    uint64 proposal_id = 1;
+}
+
+// QueryProposalResponse is the response type for the Query/Proposal RPC method.
+message QueryProposalResponse {
+    Proposal proposal = 1 [(gogoproto.nullable) = false];
+}
+
+// QueryProposalsRequest is the request type for the Query/Proposals RPC method.
+message QueryProposalsRequest {
+    option (gogoproto.equal)           = false;
+    option (gogoproto.goproto_getters) = false;
+
+    // proposal_status defines the status of the proposals.
+    ProposalStatus proposal_status = 1;
+
+    // voter defines the voter address for the proposals.
+    heimdall.types.ValidatorID voter = 2;
+
+    // depositor defines the deposit addresses from the proposals.
+    heimdall.types.ValidatorID depositor = 3;
+
+    uint64 num_limit = 4;
```

Comment:
is this pagination field or something else?

Reply:
it is not pagination field
but max num proposals to get